core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -307,6 +307,35 @@ SharedAllocationRecord<Kokkos::Experimental::HIPSpace, void>::
       "HostSpace");
 }
 
+SharedAllocationRecord<Kokkos::Experimental::HIPSpace, void>::
+    SharedAllocationRecord(
+        const Kokkos::Experimental::HIP& exec_space,
+        const Kokkos::Experimental::HIPSpace& arg_space,
+        const std::string& arg_label, const size_t arg_alloc_size,
+        const SharedAllocationRecord<void, void>::function_type arg_dealloc)
+    // Pass through allocated [ SharedAllocationHeader , user_memory ]
+    // Pass through deallocation function
+    : base_t(
+#ifdef KOKKOS_ENABLE_DEBUG
+          &SharedAllocationRecord<Kokkos::Experimental::HIPSpace,
+                                  void>::s_root_record,
+#endif
+          Kokkos::Impl::checked_allocation_with_header(arg_space, arg_label,
```

Comment:
By analogy to CUDA, do we expect to pass `exec_space` as an argument here?

Reply:
I mostly try to keep the changes here minimal. If we pass `exec_space` we need to define an `allocate` member function that takes an execution space and that one wouldn't do anything different from the other one and thus only add boilerplate.
I tend to rather change this once there exists an (asynchronous) `hipMalloc` variant taking a `hipStream_t`.

Reply:
Sounds good.